Rules.py

```diff
@@ -56,11 +56,11 @@ def set_rules(world):
                 # and render the game unbeatable, so for simplicity's sake we forbid putting pieces on any world's Song from Impa.
                 forbid_item(location, 'Triforce Piece')
 
-        if location.name == 'Forest Temple MQ First Room Chest' and world.settings.shuffle_bosskeys == 'dungeon' and world.settings.shuffle_smallkeys == 'dungeon' and world.settings.tokensanity == 'off':
-            # This location needs to be a small key. Make sure the boss key isn't placed here.
+        if location.name == 'Forest Temple MQ First Room Chest' and world.settings.shuffle_bosskeys == 'dungeon' and world.settings.shuffle_smallkeys == 'dungeon' and world.settings.tokensanity == 'off' and world.settings.logic_rules != 'none':
+            # This location needs to be a small key. Make sure the boss key isn't placed here except when logic is off.
```

Comment:
The extended comment may be unnecessary, although I'd still like for it to state it's for logical reasons rather than technical reasons.

While that location not having a key and keysanity being off would lock the player out of doing anything in forest, that seems to be what no logic players want - *true* no logic.

We're explicitly allowing wallets to lock wallets - another thing which _only_ IM will get you around - so I see no reason to not allow this chest to have something besides a key.